src/transform/CollapseTable.js

```diff
@@ -13,48 +13,25 @@ const SECTION_TOGGLED_EVENT_TYPE = 'section-toggled'
  * @return {!Array<string>}
  */
 const getTableHeader = (element, pageTitle) => {
-  let thArray = []
-
-  if (!element.children) {
-    return thArray
-  }
-
-  for (let i = 0; i < element.children.length; i++) {
-    const el = element.children[i]
-
-    if (el.tagName === 'TH') {
-      // ok, we have a TH element!
-      // However, if it contains more than two links, then ignore it, because
-      // it will probably appear weird when rendered as plain text.
-      const aNodes = el.querySelectorAll('a')
-      // todo: these conditionals are very confusing. Rewrite by extracting a
-      //       method or simplify.
-      if (aNodes.length < 3) {
-        // todo: remove nonstandard Element.innerText usage
+  const thArray = []
+  const headers = Polyfill.querySelectorAll(element, 'th')
+  if (headers) {
+    for (let i = 0; i < headers.length; ++i) {
+      const header = headers[i]
+      const anchors = Polyfill.querySelectorAll(header, 'a')
+      if (anchors.length < 3) {
         // Also ignore it if it's identical to the page title.
-        if ((el.innerText && el.innerText.length || el.textContent.length) > 0
-                && el.innerText !== pageTitle && el.textContent !== pageTitle
-                && el.innerHTML !== pageTitle) {
-          thArray.push(el.innerText || el.textContent)
+        if ((header.textContent && header.textContent.length) > 0
+          && header.textContent !== pageTitle && header.innerHTML !== pageTitle) {
```

Comment:
I think we want to use textContent but beware the differences for this conditional and the call to thArray.push() below: https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent

Reply:
Ya I read about the differences and was initially hesitant but `innerText` is just so horrible :) So far I haven't seen any ill effects...

Reply:
Guessing the combination of `innerText` forcing layouts and the unnecessarily deep recursion was the cause of the 2 order of magnitude speedup once those were removed...